.github/workflows/ubuntu.yml

```diff
@@ -28,6 +28,31 @@ jobs:
         uv sync --locked --extra dev
         uv build
 
+  test:
+    name: Ubuntu Tests
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+    - uses: actions/checkout@v5
+
+    - name: Download submodules
+      shell: bash
+      run: |
+        git submodule update --init --recursive
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version-file: "pyproject.toml"
+
+    - name: Install uv
+      uses: astral-sh/setup-uv@v6
+
+    - name: Install the project
+      run: |
+        uv sync --locked --extra dev
```

Comment:
We have already downloaded the submodules, set up python, installed uv, and build the project (drudge) during the "build" task. Should we really do them again during "test," given that "test" is dependent on (or "needs") "build"? Same for the macOS and Windows workflows.